### PR TITLE
Native apps arent always unsecure

### PR DESCRIPTION
**Pull request by isaacvetter**
Complicate the definition of public and confidential clients to more accurately account for how a native app can be a confidential client.

### CODE REVIEW

File: index.md [hunk 1]
```diff
@@ -30,30 +30,33 @@ and accounting of disclosures, are outside the scope of this profile.
...
+- OAuth 2.0 Threat Model and Security Considerations: [4.1.1. Threat: Obtaining Client Secrets](https://tools.ietf.org/html/rfc6819#section-4.1.1)
+- OAuth 2.0 for Native Apps: [8.5. Client Authentication](https://tools.ietf.org/html/draft-ietf-oauth-native-apps-12#section-8.5)
+- [OAuth 2.0 Dynamic Client Registration Protocol](https://tools.ietf.org/html/rfc7591)
 
 
+#### Use the <span class="label label-primary">confidential app</span> profile
+if your app is *able to protect* a `client_secret`, e.g.:
+
+* App runs on a trusted server with only server-side access to the secret
+* App is a native app that uses more complex technologies (such as dynamic client registration and universal redirect_uris) to protect the `client_secret`
```
Comment (gominosensei) [inline index.md]: App is a native app that uses **additional technology** (such as dynamic client registration and universal redirect_uris) to protect the `client_secret`

File: index.md [hunk 2]
```diff
@@ -30,30 +30,33 @@ and accounting of disclosures, are outside the scope of this profile.
...
+
 #### Use the <span class="label label-primary">public app</span> profile 
-when all of the following apply:
+if your app is *unable to protect* a `client_secret`, e.g.:
 
-* App runs on an end-user's device (e.g. HTML5/JS in-browser; native iOS, Windows, or Android)
-* App is *unable to protect* a `client_secret`
+* App is an HTML5/JS in-browser app that would expose the secret in user space
```
Comment (gominosensei) [inline index.md]: App is an HTML5 **or** JS in-browser app that would expose the secret in user space

**Comment by Healthedata1** [comment]
I applied this to via another PR. so can be "obviated"

**Comment by jmandel** [comment]
Specifically: fixed in https://github.com/HL7/smart-app-launch/pull/224

**Pull request closed without merging**